### PR TITLE
WIP feature plainHTTPServer please review

### PR DESCRIPTION
**Pull request by nickma82**
Hi, could someone please review my PR?

Especially a workaround the `PlainHTTPResponse()` would be nice. 
For the current state please see the working testcase which is attached as well.

Best, Nick

### CODE REVIEW

**Comment by thangbn** [comment]
thanks will take a look

**Comment by bobbui** [comment]
the test seem broken for me
```
>       self.assertTrue(expected_log_results.items() <= loads(logged_string).items())

test_plainhttpserver.py:121: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
E:\soft\sdk\python3\lib\json\__init__.py:357: in loads
    return _default_decoder.decode(s)
E:\soft\sdk\python3\lib\json\decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <json.decoder.JSONDecoder object at 0x0B95DE20>, s = '', idx = 0
```

File: tests/test_plainhttpserver.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    # https://github.com/thangbn/json-logging-python#3-configuration
+    json_logging.init_plain_http_server(enable_json=True)
+    json_logging.init_request_instrument(httpd)
+
+    logging.basicConfig(level=level)
+    json_logging.config_root_logger()
+    logging.getLogger("plain-http-request-logger").setLevel(logging.WARNING)
+
+
+class RequestHandler(BaseHTTPRequestHandler):
```
Comment (bobbui) [inline tests/test_plainhttpserver.py:24]: This should be part of the library itself. I think the idea will be user need to extend this custom requesthandler in order to extract request/response information.

**Comment by nickma82** [comment]
Hi I'm going to take a look at it soon.

File: tests/test_plainhttpserver.py (hunk 1 above)
Comment (nickma82) [reply tests/test_plainhttpserver.py:24]: I agree

**Comment by nickma82** [comment]
Won't do, sry. Best

**Pull request closed without merging**